src/vopr.zig

```diff
@@ -1028,6 +1033,7 @@ pub const Simulator = struct {
         const fault = recoverable_count >= recoverable_count_min or replica.standby();
         simulator.restart_replica(replica.replica, fault);
         maybe(!fault and replica.status == .recovering_head);
+        simulator.cluster.storages[replica.replica].faulty = true;
```

Comment:
> Whether storage faults should be enabled/disabled post restart should be at the callers' discretion.

:thinking: `restart_replica` modifies `replica_storage.faulty` already (to potentially disable faults to prevent excessive `recovering_head` replicas) -- so I would expect it to restore that, rather than expecting the caller to remember to do so.

Reply:
Mhmmmmm after all of our discussions, I _think_ I've finally understood the semantics of the `faulty` flag. I thought that setting `faulty` as `False` was something that we might want to do for the lifetime of the replica. Therefore, it felt clearer that with `restart_replica(replica, false)`, our intent was to start up a replica with storage faults disabled (and they stay disabled). 

But from the documentation as well, it's now clear we mean for this to be set transiently, only during startup.
https://github.com/tigerbeetle/tigerbeetle/blob/34e85703da17cb9b890c43855453580dfcd74c06/src/testing/storage.zig#L154-L156

I'll revert the commit that changes this behaviour.

Reply:
https://github.com/tigerbeetle/tigerbeetle/pull/2334/commits/7f7e1fda8341c4a494e13ee70d5f072d7e309919